index.js

```diff
@@ -485,7 +485,15 @@ module.exports = class Hypercore extends EventEmitter {
         }
 
         if (appended) {
-          s.emit('append')
+          let emit = false
+
+          // Always emit appends in sparse mode
+          if (s.sparse) emit = true
+          // In non-sparse mode, only emit appends if the contiguous length is
+          // also updated
+          else if (bitfield) emit = bitfield.start === s.contiguousLength
```

Comment:
Oh right, the contiguous length has already been updated if there was a bitfield present.

This could definitely use some test cases also. Given that non-sparse cores continuously replicate though, is there any way to control which ranges arrive when?

Reply:
Actually you can just do emit = tree.length === contigLength i think